Show group fines using FOLIO models and logic

marlo-longley opened the pull request.

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -1,62 +1,72 @@
 <% content_for(:navigation) do %>
-  <%= render 'shared/navigation' %>
-  <%= render partial: 'shared/group_tabs', locals: { self_value: number_to_currency(patron.fines.sum(&:owed)), group_value: number_to_currency(patron.group.fines.sum(&:owed)) } if patron_or_group.group? %>
+<%= render 'shared/navigation' %>
```
Can you restore the indent here?  It makes it easier to see that this is a block body.

jcoyne commented on app/views/summaries/_summary.html.erb:
```diff
@@ -30,15 +30,15 @@
-
-  <% if params[:group] && patron.fines.sum(&:owed).positive? # temporary fix until Symphony CGI scripts are updated %>
-    <% if patron.sponsor? # temporary fix until Symphony CGI scripts are updated %>
+  <!-- TODO: after FOLIO launch remove this condition -->
```
Make this an ERB comment so that it doesn't actually show up in the public source.

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -1,62 +1,72 @@
+    <%= sul_icon :'sharp-warning-24px' %>
+  </span>
+  Fines are accruing on <%= pluralize(accruing.length, 'overdue item') %>. You must return the
+  <%= accruing.length == 1 ? 'item' : 'items' %> before you can pay this fine.
```
Can you continue to use the pluralize helper here?

jcoyne commented on spec/views/fines/index.html.erb_spec.rb:
```diff
@@ -50,6 +59,7 @@
     end
   end
 
+  # rubocop:disable RSpec/MultipleMemoizedHelpers
```
Let's just set this threshold in rubocop.yml

jcoyne commented on config/settings/test.yml:
```diff
@@ -5,4 +5,9 @@ symws:
+  # this is a feature flag about group fines
+  # FOLIO we set it to true and because sponsor can pay all fines in their group
+  # TODO: remove after launch
+  sponsor_fines: false
```
Can this be set in `config/settings.yml` Otherwise, we get:
```
 undefined method `sponsor_fines' for nil:NilClass
```

when not using the `test` environment.

marlo-longley commented:
@jcoyne thanks for reviewing. I addressed all your comments. For the Settings file, I can make the corresponding shared_config PR as a hold for FOLIO to flip it to "true" for launch, if that seems like the right approach.

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -9,54 +9,64 @@
-        <%= render 'fines/pay_all_button' %>
-      </div>
-    <% end %>
+  <!-- TODO: after FOLIO launch remove this condition -->
```
Please use an erb comment rather than an HTML comment

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -9,54 +9,64 @@
+      contact Circulation & Privileges.
+      <% end %>
+  </div>
+  <!-- The user is a Proxy; the fine is not shown and must be paid in the Sponsor account -->
```
Please use an erb comment, not an html comment

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -9,54 +9,64 @@
+    <%= sul_icon :'sharp-error-24px', classes: 'text-recalled' %> The research group has unpaid fines that may affect
+    the status of all proxies.
+  </div>
+  <!-- The user has fines on their personal account and can pay here -->
```
Please use an erb comment, not an html comment

jcoyne commented on app/views/fines/index.html.erb:
```diff
@@ -9,54 +9,64 @@
+  </ul>
+  <% end %>
+  <% end %>
+  <!-- TODO: after FOLIO launch remove feature flag part of this condition -->
```
Please use an erb comment, not an html comment

marlo-longley commented:
@jcoyne sorry, thanks for catching those. All comments in views are now ERB comments.

marlo-longley commented:
I created the related shared_configs PR here: https://github.com/sul-dlss/shared_configs/pull/2021

cbeer commented on app/views/fines/index.html.erb:
```diff
@@ -9,54 +9,63 @@
+  <% end %>
+  <% end %>
+  <%# TODO: after FOLIO launch remove feature flag part of this condition %>
+  <% if Settings.folio.sponsor_fines && params[:group] && patron.sponsor? %>
```
Do we need a feature flag, or could we just look at what type of patron we have?
```suggestion
  <% if patron.is_a? Folio::Patron && params[:group] && patron.sponsor? %>
```

saseestone commented:
Do we need to have the same text appear for both the Proxy account view and the Sponsor account view? 

If yes, can we use:
"Fines incurred by Proxies are the responsibility of the Sponsor, and can be viewed by Sponsor's under the Self tab. Proxies can not view fines."

If we can show different text by user type, can we use:
For proxies: "Fines incurred are the responsibility of the Sponsor. They can be viewed, managed and paid only by the Sponsor."

For sponsors: Fines incurred are the responsibility of the Sponsor. They can be viewed, managed and paid under the Self tab."
